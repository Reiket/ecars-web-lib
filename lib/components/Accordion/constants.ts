export const ACCORDION_TEST_ID = 'accordionTestId';
