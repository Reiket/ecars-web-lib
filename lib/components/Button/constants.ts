import type {ButtonHTMLAttributes} from 'react';

export const BUTTON_SIZE = Object.freeze({
  BIG: 'big',
  SMALL: 'small',
});

export const BUTTON_COLOR = Object.freeze({
  GREEN: 'green',
  WHITE: 'white',
  GRAY: 'gray',
});


export type ButtonAttributesType = ButtonHTMLAttributes<HTMLButtonElement>['type']
export type ButtonSizeType = (typeof BUTTON_SIZE)[keyof typeof BUTTON_SIZE];
export type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];

export const BUTTON_TEST_ID = 'buttonTestId';