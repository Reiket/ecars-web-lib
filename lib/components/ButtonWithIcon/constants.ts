export const LEFT_ICON_BUTTON_TEST_ID = 'leftIconButtonTestId';
export const RIGHT_ICON_BUTTON_TEST_ID = 'rightIconButtonTestId';
