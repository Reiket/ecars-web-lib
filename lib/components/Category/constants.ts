export const CATEGORIES_LIST = Object.freeze({
  GUIDES: 'guides',
  REVIEWS: 'reviews',
  NEWS: 'news',
});

export type CategoriesListType = (typeof CATEGORIES_LIST)[keyof typeof CATEGORIES_LIST];

export const CATEGORY_TEST_ID = 'categoryTestId';
