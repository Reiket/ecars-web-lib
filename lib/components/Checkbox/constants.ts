import type {ElementProps} from '@/services/types';

export interface CheckboxProps extends ElementProps {
  name?: string;
  id?: string;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
  checked?: boolean;
  hasError?: boolean;
}

export const CHECKBOX_TEST_ID = 'checkboxTestId';
export const CHECKBOX_TEST_INPUT_ID = 'checkboxTestInputId';
