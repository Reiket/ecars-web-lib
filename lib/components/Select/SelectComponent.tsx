import {Select} from '@/components/Select/constants';
import type {FC} from 'react';
import {useMemo} from 'react';
import type {InputProps} from '@/components/Input/Input';

interface Props extends InputProps {
  isOpen: boolean;
  options: string[];
  onClick: () => void;
  onChange: (value: string) => void;
  value: string;
  handleSelect: (isOpen: boolean) => void;
  onClickToOptions: (value: string) => void;
  hasSearch?: boolean;
}

export const SelectComponent: FC<Props> = ({
  options,
  onClick,
  handleSelect,
  onClickToOptions,
  isOpen,
  value,
  onChange,
  block,
  id = 'select',
  name = 'select',
  disabled,
  hasSearch = false,
  ...props
}) => {
  const filteredOptions = useMemo(() => {
    if (!hasSearch || !value.trim()) {
      return options;
    }
    const filtered = options.filter((option) => option.toLowerCase().includes(value.toLowerCase()));
    return filtered.length > 0 ? filtered : null;
  }, [options, value, hasSearch]);
  const handleSelectChange = (value: string) => {
    onChange(value);
    handleSelect(hasSearch);
  };
  return (
    <Select.Wrapper
      onClickOutside={handleSelect}
      isOpen={isOpen}
      block={block}
      disabled={disabled}
    >
      <Select.Block
        onClick={onClick}
        block="select"
      >
        <Select.Input
          disabled={disabled}
          id={id}
          name={name}
          value={value}
          onChange={handleSelectChange}
          hasSearch={hasSearch}
          {...props}
        />
        <Select.Arrow />
      </Select.Block>
      <Select.List
        options={filteredOptions}
        onClick={onClickToOptions}
        value={value}
      />
    </Select.Wrapper>
  );
};
