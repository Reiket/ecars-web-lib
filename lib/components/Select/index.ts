import {SelectComponent} from '@/components/Select/SelectComponent';
import {SelectList} from '@/components/Select/components/SelectList';
import {SelectOption} from '@/components/Select/components/SelectOption';
import {SelectInput} from '@/components/Select/components/SelectInput';
import {SelectArrow} from '@/components/Select/components/SelectArrow';
import {SelectWrapper} from '@/components/Select/components/SelectWrapper';
import {SelectBlockHOC} from '@/components/Select/constants';

export const Select = Object.assign(SelectComponent, {
  List: SelectList,
  Option: SelectOption,
  Input: SelectInput,
  Arrow: SelectArrow,
  Wrapper: SelectWrapper,
  Block: SelectBlockHOC,
});
