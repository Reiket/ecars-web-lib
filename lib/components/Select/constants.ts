import {SelectBlock} from '@/components/Select/components/SelectBlock';
import {withBlockClass} from '@/services/hoc/withBlockClass';

export const SELECT_TEST_ID = 'selectTestId';

export const SelectBlockHOC = withBlockClass(SelectBlock, 'select');
