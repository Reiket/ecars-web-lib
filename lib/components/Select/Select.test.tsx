import {fireEvent, render, screen} from '@testing-library/react';
import {Select, SELECT_TEST_ID} from '@/components/Select/constants';
import {INPUT_TEST_ID} from '@/components/Input/constants';
import {selectOptionsMock} from '@/services/mocks';

const defaultProps = {
  options: selectOptionsMock,
  onClick: jest.fn(),
  handleSelect: jest.fn(),
  onClickToOptions: jest.fn(),
  onChange: jest.fn(),
  value: '',
  isOpen: true,
};

const selectOption = selectOptionsMock[0];
const optionSelector = '.select__option';
const notFoundOptionSelector = '.select__option_empty';

describe('Select Component', () => {
  const disabledOptions = [true, false];
  const placeholderOptions = ['Select', undefined];
  disabledOptions.forEach((disabled) => {
    placeholderOptions.forEach((placeholder) => {
      const disabledStatus = disabled ? 'is disabled' : 'is not disabled';
      const hasPlaceholder = placeholder ? 'has a placeholder' : 'has no placeholder';
      const testName = `should input ${disabledStatus} ${hasPlaceholder}`;
      test(testName, () => {
        const {container} = render(
          <Select
            {...defaultProps}
            disabled={disabled}
            placeholder={placeholder}
          />,
        );
        const selectElement = screen.getByTestId(SELECT_TEST_ID);
        const inputElement: HTMLInputElement = screen.getByTestId(INPUT_TEST_ID);
        if (placeholder) {
          expect(inputElement).toHaveAttribute('placeholder', placeholder);
        }
        expect(selectElement).toBeInTheDocument();
        expect(inputElement.disabled).toBe(disabled);
        if (disabled) {
          expect(selectElement).toHaveClass('disabled');
        }
        expect(container).toMatchSnapshot();
      });
    });
  });
  test('calls onChange and handleSelect when input changes', () => {
    const {container} = render(
      <Select
        {...defaultProps}
        hasSearch
      />,
    );
    const input = screen.getByTestId(INPUT_TEST_ID);
    fireEvent.change(input, {target: {value: selectOption}});
    expect(defaultProps.onChange).toHaveBeenCalledWith(selectOption);
    expect(defaultProps.handleSelect).toHaveBeenCalledWith(true);
    expect(container).toMatchSnapshot();
  });
  test('calls onClick when block is clicked', () => {
    const {container} = render(<Select {...defaultProps} />);
    const selectElement = screen.getByTestId(SELECT_TEST_ID);
    const block = selectElement.querySelector('.select__block');
    if (block) {
      fireEvent.click(block);
    }
    expect(selectElement).toHaveClass('_open');
    expect(defaultProps.onClick).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
  test('filters options when hasSearch is true and value is typed', () => {
    render(
      <Select
        {...defaultProps}
        hasSearch
        value="Opt"
      />,
    );
    const options = screen.getByTestId(SELECT_TEST_ID).querySelectorAll(optionSelector);
    expect(options.length).toBe(1);
    expect(options[0].textContent).toBe(selectOption);
  });
  test('does not render list if search yields no match', () => {
    const {container} = render(
      <Select
        {...defaultProps}
        hasSearch
        value="nonexistent"
      />,
    );
    const options = screen.getByTestId(SELECT_TEST_ID).querySelectorAll(optionSelector);
    const nullOptionsFound = screen.getByTestId(SELECT_TEST_ID).querySelector(notFoundOptionSelector);

    expect(options.length).toBe(1);
    expect(nullOptionsFound).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('closes list on outside click', () => {
    render(<Select {...defaultProps} />);
    fireEvent.mouseDown(document);
    expect(defaultProps.handleSelect).toHaveBeenCalled();
  });
  test('displays the provided value in the input and highlights the corresponding option', () => {
    const {container} = render(
      <Select
        {...defaultProps}
        value={selectOption}
      />,
    );
    const input = screen.getByTestId(INPUT_TEST_ID);
    const options = Array.from(screen.getByTestId(SELECT_TEST_ID).querySelectorAll(optionSelector));
    const activeOption = options.find((option) => option.textContent === selectOption);
    expect(input).toHaveValue(selectOption);
    expect(activeOption).toHaveClass('_selected');
    expect(container).toMatchSnapshot();
  });
  test('opens the select when typing in the input', () => {
    const {container} = render(
      <Select
        {...defaultProps}
        hasSearch
      />,
    );
    const input = screen.getByTestId(INPUT_TEST_ID);

    fireEvent.change(input, {target: {value: selectOption}});
    expect(defaultProps.isOpen).toBe(true);
    expect(container).toMatchSnapshot();
  });
});
