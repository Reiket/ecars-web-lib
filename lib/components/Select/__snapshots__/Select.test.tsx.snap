// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select Component > calls onChange and handleSelect when input changes 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component > calls onClick when block is clicked 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component > displays the provided value in the input and highlights the corresponding option 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        readonly=""
        type="text"
        value="Option 1"
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option _selected"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component > does not render list if search yields no match 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        type="text"
        value="nonexistent"
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option select__option_empty"
      >
        Nothing found
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component > opens the select when typing in the input 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component > should input is disabled has a placeholder 1`] = `
<div>
  <div
    class="select _open disabled"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        disabled=""
        id="select"
        name="select"
        placeholder="Select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component > should input is disabled has no placeholder 1`] = `
<div>
  <div
    class="select _open disabled"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        disabled=""
        id="select"
        name="select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component > should input is not disabled has a placeholder 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        placeholder="Select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component > should input is not disabled has no placeholder 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component calls onChange and handleSelect when input changes 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component calls onClick when block is clicked 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component displays the provided value in the input and highlights the corresponding option 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        readonly=""
        type="text"
        value="Option 1"
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option _selected"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component does not render list if search yields no match 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        type="text"
        value="nonexistent"
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option select__option_empty"
      >
        Nothing found
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component opens the select when typing in the input 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component should input is disabled has a placeholder 1`] = `
<div>
  <div
    class="select _open disabled"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        disabled=""
        id="select"
        name="select"
        placeholder="Select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component should input is disabled has no placeholder 1`] = `
<div>
  <div
    class="select _open disabled"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        disabled=""
        id="select"
        name="select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component should input is not disabled has a placeholder 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        placeholder="Select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;

exports[`Select Component should input is not disabled has no placeholder 1`] = `
<div>
  <div
    class="select _open"
    data-testid="selectTestId"
  >
    <div
      class="select__block"
    >
      <input
        autocomplete="off"
        class="select__input input transparent"
        data-testid="inputTestId"
        id="select"
        name="select"
        readonly=""
        type="text"
        value=""
      />
      <svg
        class="select__arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <ul
      class="select__options"
    >
      <li
        class="select__option"
      >
        Option 1
      </li>
      <li
        class="select__option"
      >
        Option 2
      </li>
      <li
        class="select__option"
      >
        Another Option
      </li>
    </ul>
  </div>
</div>
`;
